sdk/security/keyvault/azkeys/constants.go

```diff
@@ -9,244 +9,191 @@
+func PossibleSignatureAlgorithmValues() []SignatureAlgorithm {
+	return []SignatureAlgorithm{
+		SignatureAlgorithmES256,
+		SignatureAlgorithmES256K,
+		SignatureAlgorithmES384,
+		SignatureAlgorithmES512,
+		SignatureAlgorithmPS256,
+		SignatureAlgorithmPS384,
+		SignatureAlgorithmPS512,
+		SignatureAlgorithmRS256,
+		SignatureAlgorithmRS384,
+		SignatureAlgorithmRS512,
+		SignatureAlgorithmRSNULL,
 	}
 }
 
-// JSONWebKeyType - JsonWebKey Key Type (kty), as defined in https://tools.ietf.org/html/draft-ietf-jose-json-web-algorithms-40.
-type JSONWebKeyType string
+// KeyType - JsonWebKey Key Type (kty), as defined in https://tools.ietf.org/html/draft-ietf-jose-json-web-algorithms-40.
+type KeyType string
 
 const (
-	// JSONWebKeyTypeEC - Elliptic Curve.
-	JSONWebKeyTypeEC JSONWebKeyType = "EC"
-	// JSONWebKeyTypeECHSM - Elliptic Curve with a private key which is stored in the HSM.
-	JSONWebKeyTypeECHSM JSONWebKeyType = "EC-HSM"
-	// JSONWebKeyTypeOct - Octet sequence (used to represent symmetric keys)
-	JSONWebKeyTypeOct JSONWebKeyType = "oct"
-	// JSONWebKeyTypeOctHSM - Octet sequence (used to represent symmetric keys) which is stored the HSM.
-	JSONWebKeyTypeOctHSM JSONWebKeyType = "oct-HSM"
-	// JSONWebKeyTypeRSA - RSA (https://tools.ietf.org/html/rfc3447)
-	JSONWebKeyTypeRSA JSONWebKeyType = "RSA"
-	// JSONWebKeyTypeRSAHSM - RSA with a private key which is stored in the HSM.
-	JSONWebKeyTypeRSAHSM JSONWebKeyType = "RSA-HSM"
+	// KeyTypeEC - Elliptic Curve.
+	KeyTypeEC KeyType = "EC"
+	// KeyTypeECHSM - Elliptic Curve with a private key which is stored in the HSM.
+	KeyTypeECHSM KeyType = "EC-HSM"
+	// KeyTypeOct - Octet sequence (used to represent symmetric keys)
+	KeyTypeOct KeyType = "oct"
+	// KeyTypeOctHSM - Octet sequence (used to represent symmetric keys) which is stored the HSM.
+	KeyTypeOctHSM KeyType = "oct-HSM"
+	// KeyTypeRSA - RSA (https://tools.ietf.org/html/rfc3447)
+	KeyTypeRSA KeyType = "RSA"
+	// KeyTypeRSAHSM - RSA with a private key which is stored in the HSM.
+	KeyTypeRSAHSM KeyType = "RSA-HSM"
 )
 
-// PossibleJSONWebKeyTypeValues returns the possible values for the JSONWebKeyType const type.
-func PossibleJSONWebKeyTypeValues() []JSONWebKeyType {
-	return []JSONWebKeyType{
-		JSONWebKeyTypeEC,
-		JSONWebKeyTypeECHSM,
-		JSONWebKeyTypeOct,
-		JSONWebKeyTypeOctHSM,
-		JSONWebKeyTypeRSA,
-		JSONWebKeyTypeRSAHSM,
+// PossibleKeyTypeValues returns the possible values for the KeyType const type.
+func PossibleKeyTypeValues() []KeyType {
+	return []KeyType{
+		KeyTypeEC,
+		KeyTypeECHSM,
+		KeyTypeOct,
+		KeyTypeOctHSM,
+		KeyTypeRSA,
+		KeyTypeRSAHSM,
 	}
 }
 
-// KeyEncryptionAlgorithm - The encryption algorithm to use to protected the exported key material
-type KeyEncryptionAlgorithm string
+// KeyExportEncryptionAlgorithm - The encryption algorithm to use to protected the exported key material
+type KeyExportEncryptionAlgorithm string
```

Comment:
I like the old name better because these are simply algorithms for encrypting keys. They aren't algorithms specifically for encrypting keys for export, though that may be Key Vault's only use of them today.

Reply:
Added the Export part to help differentiate it from type `Encryption algorithm`

Reply:
But **Key**EncryptionAlgorithm is already different and model definitions make it obvious where to use each type

Reply:
Keeping as `KeyEncryptionAlgorithm`. Fixed in [6eb8c22](https://github.com/Azure/azure-sdk-for-go/pull/20892/commits/6eb8c22e41825a190c2df4fdd2f2361e26f93280)

Reply:
I missed the context here, but in other languages we call this `KeyWrapAlgorithm` because it coincides with the `WrapKey` method. It just sounded better the other way.